lib/staging/gpio/gpio.cpp

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2023 Pionix GmbH and Contributors to EVerest
+
+#include "gpio.hpp"
+
+#include <system_error>
+
+namespace Everest {
+
+bool Gpio::open(const std::string& chip_name, int line_number, bool _inverted) {
+
+    if (chip_name == "") {
+        is_open = false;
+    } else {
+        try {
+            chip.open(chip_name);
+            line = chip.get_line(line_number);
+            is_open = true;
+        } catch (const std::system_error& e) {
+            is_open = false;
+        } catch (const std::out_of_range& e) {
+            is_open = false;
+        }
+    }
+
+    inverted = _inverted;
```

Comment:
We can consider `this` keyword here

Reply:
we could use this->inverted=inverted here, I think it is a matter of personal taste :-) I personally don't like shadowing member variables with local variables that much as it always causes confusion.